src/pages/waiting-room/cat-home-test/__tests__/waiting-room.cat-d.page.spec.ts

```diff
@@ -0,0 +1,291 @@
+      it('should emit an insurance declaration toggle action when changed', () => {
+        component.insuranceDeclarationChanged();
+
+        expect(store$.dispatch).toHaveBeenCalledWith(new ToggleInsuranceDeclaration());
+      });
+    });
+
+    describe('dispatchCandidateChoseToProceedInWelsh', () => {
+      it('it should dispatch CandidateChoseToProceedWithTestInWelsh action', () => {
+        component.dispatchCandidateChoseToProceedInWelsh();
+        expect(store$.dispatch)
+          .toHaveBeenCalledWith(new communicationPreferenceActions.CandidateChoseToProceedWithTestInWelsh(
+            Language.CYMRAEG));
+      });
+    });
+
+    describe('dispatchCandidateChoseToProceedInEnglish', () => {
+      it('it should dispatch CandidateChoseToProceedWithTestInEnglish action', () => {
+        component.dispatchCandidateChoseToProceedInEnglish();
+        expect(store$.dispatch)
+          .toHaveBeenCalledWith(new communicationPreferenceActions.CandidateChoseToProceedWithTestInEnglish(
+            Language.ENGLISH));
+      });
+    });
+
+    describe('ionViewDidEnter', () => {
+      it('should enable single app mode if on ios', () => {
+        component.ionViewDidEnter();
+        expect(deviceProvider.enableSingleAppMode).toHaveBeenCalled();
+      });
+      it('should lock the screen orientation to Portrait Primary', () => {
+        component.ionViewDidEnter();
+        expect(screenOrientation.lock)
+          .toHaveBeenCalledWith(screenOrientation.ORIENTATIONS.PORTRAIT_PRIMARY);
+      });
+
+      it('should keep the device awake', () => {
+        component.ionViewDidEnter();
+        expect(insomnia.keepAwake).toHaveBeenCalled();
+      });
+
+    });
+
+    describe('clickBack', () => {
+      it('should should trigger the lock screen', () => {
+        component.clickBack();
+        expect(deviceAuthenticationProvider.triggerLockScreen).toHaveBeenCalled();
+      });
+    });
+
+    describe('onSubmit', () => {
+      it('should navigate to the COMMUNICATION_PAGE if the form is valid', () => {
+        const formGroup = component.formGroup;
+        formGroup.addControl('insuranceCheckbox', new FormControl('', [Validators.requiredTrue]));
+        formGroup.get('insuranceCheckbox').setValue(true);
+        component.onSubmit();
+        expect(navController.push).toHaveBeenCalled();
+      });
+      it('should dispatch the WaitingRoomValidationError action if a field is not valid', fakeAsync(() => {
+        const formGroup = component.formGroup;
+        formGroup.addControl('insuranceCheckbox', new FormControl('', [Validators.requiredTrue]));
+        formGroup.get('insuranceCheckbox').setValue(false);
+        component.onSubmit();
+        tick();
+        expect(store$.dispatch).toHaveBeenCalledWith(new WaitingRoomValidationError('insuranceCheckbox is blank'));
+      }));
+    });
+
+    describe('isJournalDataInvalid', () => {
+      const journalData: CatDUniqueTypes.JournalData = {
```

Comment:
to-do missing

Reply:
instead of a TODO - it now uses the right type